compiler/ballerina-parser/src/test/resources/expressions/on-fail-check/on_fail_check_source_09.bal

```diff
@@ -0,0 +1,5 @@
+function foo() {
+    _ = from int a in b
+        join C d in e on f equals g
```

Comment:
My bad.
``` suggestion
        join C d in check e on f equals g
```

Change file name to `check_expr_source` and move out of recovery section